### ISSUE TITLE
NTItemParser Limitation

### ISSUE THREAD

**Issue by KevinTyrrell** [kolton/d2bot-with-kolbot#612]
Items like [The Rising Sun](http://diablo.wikia.com/wiki/The_Rising_Sun) and [Snowclash](http://diablo.wikia.com/wiki/Snowclash) have affixes such as `x% chance to cast Level y <Spell Name> on <Event>`. Currently there is no way to specify in the `nip` pickit files either the `x`, `y` or `<Spell Name>` field. This is strange to me though, because on the Project  ETAL forums, [they claim that D2BS has support for this](https://www.projectetal.com/forums/threads/snowclash-when-hit.32941/). I tried their solution of a `description` field but unfortunately after looking at the source code, there is nothing inside of [NTItemParser](https://github.com/kolton/d2bot-with-kolbot/blob/master/d2bs/kolbot/libs/NTItemParser.dbl) that can process such a request.

`Unknown stat: description`

I suggest a new property tag should be introduced for this edge case. As a developer I'm not a fan with the `description` idea, since it's expensive for String comparisons (O(N) complexity) on every item.

**Comment by gogos-venge** [comment]
That's a nice question.
Since you are a coder, you should already know that in order to keep things from getting REALLY weird, sometimes we sacrifice a small portion of optimization. I mean c'mon, I am running this bot on a 4790k... How bad is a string comparison to that CPU? I wouldn't care less.
But since you asked, yes it is possible, since the hook has a function exposed for reading item stats and also their subindex.
Although it's somewhat difficult to use. For example in order to check The Rising Sun, you could call the method getStat(201,3603) on the item object (201 is stat id and 3603 [or 0xE13] is subindex) in order to get the % *CHANCE* as an integer to cast **Level 19 Meteor** when being struck. The **EVENT** values are already set inside **NtItemAlias.dbl** and the one I mention, 201, is at line 1297. Now if you wanna get the % chance to cast Level **18** Meteor when struck you need to call again the method getStat(201,3602) on the item object but as you see the subindex is 3602 (or 0xE12) this time. And this will give you 2 again since The Rising Sun always has a 2% chance to cast a variable level Meteor when struck. Now one way to get the Meteor level is to actually get the array of stats from the item with getStat(-1) and search for 201. The NEXT item in the array is the subindex of it. I think (please correct me if I'm wrong) the bitwise of the subindex & 0xFF will give you the skill level. In this case of Meteor 19 level: 0xE13 & 0xFF = 0x13 which is 19 in decimal, which is the Meteor level. This is fucked up shit.

**Comment by mf022** [comment]
> Items like and Snowclash have affixes such as x% chance to cast Level y <Spell Name> on <Event>

[name] == battlebelt && [quality] == unique # [enhanceddefense] >= 170  && [description] == " level 19 " 
isn't working for [snowclash](http://classic.battle.net/diablo2exp/items/exceptional/ubelts.shtml)?

**Comment by gogos-venge** [comment]
@mf022 he said he doesn't want to make string comparisons, as he finds it inefficient.

**Comment by KevinTyrrell** [comment]
@mf022 No. As stated above if you attempt to do that you will receive the following: `Unknown stat: description`.
